### PR TITLE
Alternative C++ Solution to Silver Bovine Shuffle (usaco-764.mdx) + Complete Search With Recursion Problem Suggestion

### PR DESCRIPTION
**Pull request by joshualiu555**

### CODE REVIEW

File: solutions/usaco-764.mdx [hunk 1]
```diff
@@ -80,6 +80,68 @@ int main(){
...
+   }
+   if (in_cycle) {
+       ans[node] = node;
+   } else {
+       ans[node] = ans[p[node]];
+   }
+}
+
+int main()
+{
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:119]: please have a consistent brace style

File: solutions/usaco-764.mdx [hunk 2]
```diff
@@ -80,6 +80,68 @@ int main(){
...
+<CPPSection>
+	
+## Alternative Explanation
+	
+This problem can actually be solved using the first answer to “Badge.” In that question, the nodes whose answer was itself were the ones that were in a cycle. Otherwise, it would take a few steps to reach a cycle. If a node is in a cycle, it will never be empty. Otherwise, it will at some point be empty. You can literally just copy and paste the previous solution. Then, count how many nodes’ answer are itself. 
+	
+## Alternative Implementation
+	
+```cpp
+int n, p[INF], ans[INF];
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:94]: comments explaining what these vars mean please?
or just better var names?

File: solutions/usaco-764.mdx [hunk 3]
```diff
@@ -80,6 +80,68 @@ int main(){
...
+bool in_cycle = false;
+
+void dfs(int node) {
+   if (ans[node] != -2) {
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:98]: comments would be kinda nice

File: solutions/usaco-764.mdx [hunk 4]
```diff
@@ -80,6 +80,68 @@ int main(){
...
+
+   for (int i = 0; i < n; i++) dfs(i);
+  
+   int count = 0;
+   for (int i = 0; i < n; i++) {
+       if (ans[i] == i) count++;
+   }
+   cout << count << '\n';
+
+   return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:141]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-764.mdx (hunk 3 above)
Comment (joshualiu555) [reply solutions/usaco-764.mdx:98]: Sorry about that. I'll continue to fix all the issues.

**Comment by joshualiu555** [comment]
Fixed inconsistent brace style, explained variable meanings, added comments

File: solutions/usaco-764.mdx [hunk 5]
```diff
@@ -80,6 +80,84 @@ int main(){
...
+    else {
+        cycle[node] = cycle[p[node]];
+    }
+}
+
+int main() {
+    std::ios_base::sync_with_stdio(false); cin.tie(0);
+
+//    ifstream cin("shuffle.in");
+//    ofstream cout("shuffle.out");
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:138]: wait what

Comment (joshualiu555) [reply solutions/usaco-764.mdx:138]: It's just the IO

Comment (joshualiu555) [reply solutions/usaco-764.mdx:138]: Should I uncomment it?

Comment (SansPapyrus683) [reply solutions/usaco-764.mdx:138]: uh yeah

File: solutions/usaco-764.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-764.mdx]: bump

File: solutions/usaco-764.mdx (hunk 5 above)
Comment (joshualiu555) [reply solutions/usaco-764.mdx]: Fixed

File: solutions/usaco-764.mdx [hunk 6]
```diff
@@ -80,6 +80,84 @@ int main(){
...
+    for (int i = 0; i < n; i++) dfs(i);
+
+    int ans = 0;
+    for (int i = 0; i < n; i++) {
+        // if this node cycles back to itself, it means it is never empty
+        if (cycle[i] == i) ans++;
+    }
+    cout << ans << '\n';
+
+    return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:157]:
```suggestion
```

File: solutions/usaco-764.mdx [hunk 7]
```diff
@@ -2,7 +2,7 @@
 id: usaco-764
 source: USACO Silver 2017 December
 title: The Bovine Shuffle
-author: Jesse Choe, Tanish Tyagi
+author: Jesse Choe, Tanish Tyagi, Joshua Liu
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_shuffle_silver_dec17.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-764.mdx:8]:
```suggestion
[Official Analysis (Java)](http://www.usaco.org/current/data/sol_shuffle_silver_dec17.html)
```

File: content/2_Bronze/Complete_Rec.problems.json [hunk 8]
```diff
@@ -97,6 +97,19 @@
...
+      "uniqueId": "usaco-692",
+      "name": "Secret Cow Code",
+      "url": "http://www.usaco.org/index.php?page=viewproblem2&cpid=692",
+      "source": "Silver",
+      "difficulty": "Insane",
+      "isStarred": false,
+      "tags": ["Recursion", "Ad Hoc"],
+      "solutionMetadata": {
+        "kind": "USACO",
+        "usacoId": "692"
```
Comment (ryanchou-dev) [inline content/2_Bronze/Complete_Rec.problems.json:111]:
```suggestion
        "kind": "internal"
```
supposed to be like this for internal sols?

File: solutions/usaco-764.mdx [hunk 9]
```diff
@@ -80,6 +80,82 @@ int main(){
...
+<CPPSection>
+	
+## Alternative Explanation
+	
+This problem can actually be solved using the first answer to “Badge.” In that question, the nodes whose answer was itself were the ones that were in a cycle. Otherwise, it would take a few steps to reach a cycle. If a node is in a cycle, it will never be empty. Otherwise, it will at some point be empty. You can literally just copy and paste the previous solution. Then, count how many nodes’ answer are itself. 
+	
+## Alternative Implementation
+	
+```cpp
+// Created By Joshua Liu
```
Comment (ryanchou-dev) [inline solutions/usaco-764.mdx:94]:
```suggestion
```
name already in authors

File: solutions/usaco-764.mdx [hunk 10]
```diff
@@ -80,6 +80,82 @@ int main(){
...
+        p[i]--;
+    }
+    
+    fill(cycle, cycle + n, -2);
+
+    for (int i = 0; i < n; i++) dfs(i);
+
+    int ans = 0;
+    for (int i = 0; i < n; i++) {
+        // if this node cycles back to itself, it means it is never empty
```
Comment (ryanchou-dev) [inline solutions/usaco-764.mdx:152]:
```suggestion
        // if this node cycles back to itself, it means it's never empty
```
optional

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
cool

**Comment by ryanchou-dev** [comment]
why was this closed?

**Pull request closed without merging**